pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+    move_df = _default_move_df()
+
+    move_df = move_df.loc[1:]
+    assert_array_equal(move_df.index, [1, 2, 3])
+    move_df.reset_index(inplace=True)
+    assert_array_equal(move_df.index, [0, 1, 2])
+
+
+def test_set_index():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05')],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06')],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11')],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11')],
+    ], columns=['lat', 'lon', 'datetime'], index=[1, 1, 2, 2]
+    )
+    expected.index.name = 'id'
+    assert_frame_equal(move_df.set_index('id'), expected)
+
+    try:
+        move_df.set_index('datetime', inplace=True)
+        assert False
+    except AttributeError:
+        assert True
+
+
+def test_drop():
+    move_df = _default_move_df()
+    move_df[UID] = [1, 1, 2, 3]
+
+    move_test = move_df.drop(columns=[UID])
+    assert UID not in move_test
+    assert UID in move_df
+    assert isinstance(move_test, PandasMoveDataFrame)
+
+    move_test = move_df.drop(index=[0, 1])
+    assert move_test.len() == 2
+    assert isinstance(move_test, PandasMoveDataFrame)
+
+    move_df.drop(columns=[UID], inplace=True)
+    assert UID not in move_df
+    assert isinstance(move_df, PandasMoveDataFrame)
+
+    try:
+        move_df.drop(columns=[LATITUDE], inplace=True)
+        raise AssertionError('AttributeError error not raised by MoveDataFrame')
+    except AttributeError:
+        pass
+
+    try:
+        move_df.drop(columns=[LONGITUDE], inplace=True)
+        raise AssertionError('AttributeError error not raised by MoveDataFrame')
+    except AttributeError:
+        pass
+
+    try:
+        move_df.drop(columns=[DATETIME], inplace=True)
+        raise AssertionError('AttributeError error not raised by MoveDataFrame')
+    except AttributeError:
+        pass
+
+
+def test_duplicated():
+    move_df = _default_move_df()
+
+    expected = [False, True, False, True]
+    assert_array_equal(move_df.duplicated(TRAJ_ID), expected)
+    expected = [False, False, True, False]
+    assert_array_equal(move_df.duplicated(subset=DATETIME, keep='last'), expected)
+
+
+def test_drop_duplicates():
+    move_df = _default_move_df()
+
+    move_test = move_df.drop_duplicates()
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[0, 1, 2]
+    )
+    assert_frame_equal(move_test, expected)
+    assert isinstance(move_test, PandasMoveDataFrame)
+    assert move_df.len() == 4
+
+    move_df.drop_duplicates(inplace=True)
+    assert_frame_equal(move_test, expected)
+    assert isinstance(move_df, PandasMoveDataFrame)
+    assert move_df.len() == 3
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)